### DISCUSSION TITLE
Why remaking drag-and-drop alpinejs example I lost some functionality?

### DISCUSSION THREAD

**Discussion by sergeynilov** [alpinejs/alpine#1223] [1. Help]
Hello,
With AlpineJs 2.8 I try to remake drag-and-drop example from
https://codepen.io/trovster/pen/oNjGGMq

based on dynamic data, but not all works as expect at :
https://codepen.io/sergeynilov/pen/zYNobRP

When I drag element from left to right, I still have element at left panel
 (it must be hidden)

I remade block :
```
                        <template x-for="nextUser in users" :key="nextUser">

                            <li
                                :id="'item-' + nextUser.id"
                                class="drag-and-drop__item"
                                :class="{ 'drag-and-drop__item--dragging': dragging }"

```
trying to make any element with unique ID, but looks like that is not enough...
What Did I miss ?

Thanks!

**Comment by KevinBatdorf** [comment]
You're moving the node from one side to another, but you're not changing the Alpine component data, which will re-render on mutation and show the complete list again. You need to keep track of what is moved and remove it from the `users` list. 

You probably want to keep two lists, one for each list and manage both together, checking the consistency when you drop an item. And remove the imperative append code that fires on drop.

Also, `:key="nextUser"` you want to use a primitive type, not the object. Change this to `:key="nextUser.id"`
